lib/imports.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const {getOriginal} = require('./originals-helper.js');
+
+/**
+ * Formats the information into the right format for import
+ *
+ * @param {string} identifier - the name of the method / function to get
+ * @param {string} type - the type of original of the identifier
+ * @param {string} mod - the name of the module to import from
+ *
+ * @return {string} the formatted import text
+ */
+function generateImportText(identifier, type, mod) {
+  const original = getOriginal(identifier, type, mod);
+  switch (type) {
+    case 'constructor':
+      return original;
+    case 'method':
+      return `${identifier} as ${original}`;
+    default:
+      throw new Error('not yet implemented');
+  }
+}
+
+/**
+ * Adds an import for the module to the list
+ *
+ * @param {Object} imports - the current list of imports
+ * @param {string} mod - the name of the module to import from
+ * @param {string} identifier - the name of the method / function to get
+ * @param {string} type - the type of original of the identifier
+ */
+function updateImports(imports, mod, identifier, type) {
+  const importText = generateImportText(identifier, type, mod);
+
+  if (!(mod in imports)) {
+    imports[mod] = {
+      namedImports: [],
+    };
+  }
+
+  switch (type) {
+    case 'constructor':
+      imports[mod].defaultImport = importText;
+      return;
+    case 'method':
+      if (!imports.Reflect) {
```

Comment:
It might be simpler if you made namedImports a Set. Then you could just add() to it and not worry about if it already exists (because adding something that already exists to a Set is a no-op). This might require converting it to an array (with `[...set]`) before passing it to ts-morph, but I think it would clean up some of the code here a decent bit.